test/GraphingSpec.hs

```diff
@@ -196,3 +264,117 @@ spec = do
+pruneUnreachableSpec =
+  describe "pruneUnreachable" $ do
+    it "A graph with no direct nodes should be empty after prune" $ do
+      let graph :: Graphing Int
+          graph = Graphing.deeps [1, 2, 3] <> Graphing.edges [(1, 2), (3, 2)]
+      Graphing.pruneUnreachable graph `shouldBe` mempty
+
+    it "A graph should remove unreachable nodes" $ do
+      -- 1 -> 2 -> 3
+      -- 4
+      --           5 (deep)
+      let graph :: Graphing Int
+          graph =
+            Graphing.directs [1, 4]
+              <> Graphing.edges [(1, 2), (2, 3), (5, 5)]
+              <> Graphing.deep 5
+          pruned = Graphing.pruneUnreachable graph
+
+      expectDeps [1, 2, 3, 4] pruned
+      expectEdges [(1, 2), (2, 3)] pruned
+
+shrinkRootsSpec :: Spec
+shrinkRootsSpec =
+  describe "shrinkRoots" $ do
+    it "should remove direct nodes" $ do
+      let graph :: Graphing Int
+          graph = Graphing.directs [1] <> Graphing.edges [(1, 2), (1, 3), (2, 4), (3, 6)]
+          graph' = Graphing.shrinkRoots graph
+
+      expectDirect [2, 3] graph'
+      expectDeps [2, 3, 4, 6] graph'
+      expectEdges [(2, 4), (3, 6)] graph'
+
+    it "should not modify when there are no direct nodes" $ do
+      let graphWithoutDirectNodes :: Graphing Int
+          graphWithoutDirectNodes = Graphing.edges [(1, 2), (1, 3), (2, 4), (3, 6)]
+
+          graph' :: Graphing Int
+          graph' = Graphing.shrinkRoots graphWithoutDirectNodes
+
+      expectDirect [] graph'
+      expectDeps [1, 2, 3, 4, 6] graph'
+      expectEdges [(1, 2), (1, 3), (2, 4), (3, 6)] graph'
+
+splitGraphSpec :: Spec
+splitGraphSpec =
+  describe "splitGraph" $ do
+    it "Returns Nothing for a graphing with a cycle" $ do
+      let am :: AM.AdjacencyMap Int
+          am = AM.edge 1 1
+      splitGraph am `shouldBe` Nothing
+    it "Should split graphs into unconnected subgraphs" $ do
+      let graph :: AM.AdjacencyMap Int
+          graph = AM.edges [(1, 2), (2, 3), (4, 5)]
+          subGraph1 = AM.edges [(1, 2), (2, 3)]
+          subGraph2 = AM.edge 4 5
+      maybe (fail "") (`shouldMatchList` [subGraph1, subGraph2]) (splitGraph graph)
+
+stripRootSpec :: Spec
+stripRootSpec =
+  describe "stripRoot" $ do
+    let graph :: Graphing Int
+        graph = Graphing.directs [1] <> Graphing.edges [(1, 2), (1, 3), (2, 4), (3, 6)]
+        graph' = Graphing.stripRoot graph
+
+    it "should promote immediate children as direct nodes" $ do
+      expectDirect [2, 3] graph'
+
+    it "should preserve current root nodes in the graphing as nodes" $ do
+      expectDeps [1, 2, 3, 4, 6] graph'
+
+    it "should preserve edges of current root nodes in the graphing" $ do
+      expectEdges [(1, 2), (1, 3), (2, 4), (3, 6)] graph'
+
+spec :: Spec
+spec = do
+  unfoldSpec
+
+  accessingElementsSpec
+
+  addingNodesSpec
+
+  shrinkingSpec
+
+  pruneUnreachableSpec
+
+  getRootsOfSpec
+
+  shrinkRootsSpec
+
+  splitGraphSpec
+
+  stripRootSpec
+
+  describe "promoteToDirect" $ do
```

Comment:
Why not move these to its own spec items as you have done for other ones?

Reply:
I thought they were maybe too small, but I've moved them now.